force-app/main/default/classes/PS_IntegrationServiceConfig.cls

```diff
@@ -72,28 +75,64 @@ public with sharing class PS_IntegrationServiceConfig {
         SFDO_GATEWAYIDS
     };
 
+    /**
+    * @description Returns the Elevate integration service enablement status
+    * @return Boolean
+    */
+    public Boolean isIntegrationEnabled() {
+        return isEnabled;
+    }
+
+    /**
+    * @description Determines if the current user has the appropriate permissions to modify Elevate records
+    * @return Boolean
+    */
+    public Boolean hasIntegrationPermissions() {
+        return hasPermissions;
+    }
+
     /**
     * @description Determines if the Elevate integration service is configured and enabled.
-    * The Elevate Payment Services integration is considered "enabled" (for now) when there
-    * is at least a single record in the protected Payment Services Configuration object
+    * The Elevate Payment Services integration is considered enabled when all required
+    * service records are present with value and isActive record is either missing or true
     * @return Boolean
     */
     private Boolean isEnabled {
         get {
             if (isEnabled == null) {
-                return config.keyValueMap.keySet().containsAll(REQUIRED_CONFIG_KEYS);
+                return isIntegrationActive() && hasElevateServiceData();
```

Comment:
Is there additional value to having these two method calls in a private getter? Could we just call these two private methods directly in the public `isIntegrationEnabled`?